provider/oci/userdata.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package oci
+
+import (
+	b64 "encoding/base64"
+
+	"github.com/juju/errors"
+	jujuos "github.com/juju/utils/os"
+
+	"github.com/juju/juju/cloudconfig/cloudinit"
+	"github.com/juju/juju/cloudconfig/providerinit/renderers"
+)
+
+// OCIRenderer implements the renderers.ProviderRenderer interface
+type OCIRenderer struct{}
+
+// Renderer is defined in the renderers.ProviderRenderer interface
+func (OCIRenderer) Render(cfg cloudinit.CloudConfig, os jujuos.OSType) ([]byte, error) {
+	var renderedUdata []byte
+	var err error
+	switch os {
+	case jujuos.Ubuntu, jujuos.CentOS:
```

Comment:
Does it really make sense to hard code ubuntu and CentOS here?

Reply:
It is how it is done in other provider renderers such as:
```
func (AmazonRenderer) Render(cfg cloudinit.CloudConfig, os jujuos.OSType) ([]byte, error) {
	switch os {
	case jujuos.Ubuntu, jujuos.CentOS:
		return renderers.RenderYAML(cfg, utils.Gzip)
	case jujuos.Windows:
		return renderers.RenderYAML(cfg, renderers.WinEmbedInScript, renderers.AddPowershellTags)
	default:
		return nil, errors.Errorf("Cannot encode userdata for OS: %s", os.String())
	}
}
```